### PR TITLE
change avatar and name

### PR DESCRIPTION
**Pull request by ghost**
Added bot commands and api endpoints to allow people to create separate instances of the main bot for customization,
Avatar and name can be changes through bot commands, but new instances can only be created through the dashboard currently.

One instance can be created per guild, one person can manage multiple instances.
The custom instances can be invited to guilds other than the one it was created for, but it will need to be managed from the main guild.

### CODE REVIEW

**Comment by ghost** [comment]
dashboard pull request: https://github.com/CreatorCoinTools/rally-discordbot-dashboard/pull/5

File: rallyrolebot/api/bot_activity_mappings.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+router = APIRouter(
+    prefix="/mappings/bot_activity",
+    tags=["bot_avatar"],
+    dependencies=[Depends(owner_or_admin)],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/{guildId}", response_model=BotActivityMapping)
+async def read_mapping(guildId):
```
Comment (Ju99ernaut) [inline rallyrolebot/api/bot_activity_mappings.py]: declare type for `guildId`

File: rallyrolebot/api/bot_activity_mappings.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        return {}
+
+    activity_text = bot_instance[BOT_ACTIVITY_TEXT_KEY]
+    activity_type = bot_instance[BOT_ACTIVITY_TYPE_KEY]
+
+    return {"activity_text": activity_text, "activity_type": activity_type}
+
+
+@router.post("", response_model=BotActivityMapping)
+async def add_mapping(mapping: BotActivityMapping, guildId):
```
Comment (Ju99ernaut) [inline rallyrolebot/api/bot_activity_mappings.py]: declare type for `guildId`

File: rallyrolebot/api/bot_avatar_mappings.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+router = APIRouter(
+    prefix="/mappings/bot_avatar",
+    tags=["bot_avatar"],
+    dependencies=[Depends(owner_or_admin)],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/{guildId}", response_model=BotAvatarMapping)
+async def read_mapping(guildId):
```
Comment (Ju99ernaut) [inline rallyrolebot/api/bot_avatar_mappings.py]: declare type for `guildId`

File: rallyrolebot/api/bot_avatar_mappings.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    if not avatar:
+        bot_object = update_cog.running_bots[bot_instance[BOT_ID_KEY]]['bot']
+        avatar = bot_object.user.avatar_url
+        data.set_bot_avatar(guildId, str(bot_object.user.avatar_url))
+
+    return {"bot_avatar": avatar, "avatar_timeout": int(bool(bot_instance[AVATAR_TIMEOUT_KEY])), "guildId": guildId}
+
+
+@router.post("", response_model=BotAvatarMapping)
+async def add_mapping(mapping: BotAvatarMapping, guildId):
```
Comment (Ju99ernaut) [inline rallyrolebot/api/bot_avatar_mappings.py]: declare type for `guildId`

**Review by Ju99ernaut — CHANGES_REQUESTED** [review]
Well everything looks good, really great work,  just a few really small changes. Anyways what are the implications on performance with this approach?

**Comment by ghost** [comment]
> Anyways what are the implications on performance with this approach?

As far as I can tell it should be very small, its all running on the same stack so it shouldn't take up that much more resources. I tried running 10 bot instances and I didn't see any hit to performance.
The bot instances are just different interfaces activating the same code on the same stack, should be all good

**Review by Ju99ernaut — APPROVED** [review]

**Pull request merged**